.ci-dockerfiles/centos7/README.md

```diff
@@ -0,0 +1,32 @@
+# Build image
+
+```bash
+docker build -t ponylang/ponyc-ci:centos7 .
+```
+
+# Run image to test
+
+Will get you a bash shell in the image to try cloning Pony into where you can test a build to make sure everything will work before pushing:
+
+```bash
+docker run --name ponyc-ci-centos7 --user pony --rm -i -t ponylang/ponyc-ci:centos7 bash
+```
+
+## Run CircleCI jobs locally
+
+Use the [CircleCI CLI](https://circleci.com/docs/2.0/local-cli/) to run the CI job using this image
+from the ponyc project root:
+
+```bash
+circleci build --job centos7-debug
+circleci build --job centos7-release
+circleci build --job centos7-llvm-600-release
```

Comment:
this is incorrect

Reply:
I did add centos-llvm-600-release to .circleci/config.yml, so I think the above is correct.

Do you mean its incorrect because you don't want to support llvm 6.0.0? The reason I kept support for llvm 6.0.0 is becuase I assume that sooner or later it will be supported. Also, there is already a significant support for llvm > 3.9.1, see below, and that if we don't test that code it will become stale.

Also, we probably want to remove support for llvm > 3.9.1 and < 6.0.0 after add hosting llvm, if you agree I'll create an issue to remove unsupported llvm code.
```
$ ack PONY_LLVM
src/libponyc/codegen/genoperator.c
366:#if PONY_LLVM >= 400

src/libponyc/codegen/genjit.cc
22:#if PONY_LLVM >= 500
35:#if PONY_LLVM >= 500
75:#if PONY_LLVM >= 500
113:#if PONY_LLVM >= 500
121:#if PONY_LLVM >= 400
132:#if PONY_LLVM >= 500
137:#if PONY_LLVM >= 500
140:#  if PONY_LLVM >= 400
154:#if PONY_LLVM >= 500

src/libponyc/codegen/gendebug.cc
25:#if PONY_LLVM < 500
37:#if PONY_LLVM < 500
72:#if PONY_LLVM < 600
86:#if PONY_LLVM >= 400
111:#if PONY_LLVM >= 400
112:#  if PONY_LLVM >= 500
143:#if PONY_LLVM >= 400
166:#if PONY_LLVM >= 400
181:#if PONY_LLVM >= 400
210:#if PONY_LLVM >= 400
244:#if PONY_LLVM >= 400
272:#if PONY_LLVM >= 400

src/libponyc/codegen/gendesc.c
74:#if PONY_LLVM >= 400
492:#if PONY_LLVM >= 400
506:#if PONY_LLVM >= 400
545:#if PONY_LLVM >= 400
581:#if PONY_LLVM >= 400
672:#if PONY_LLVM >= 400

src/libponyc/codegen/codegen.c
19:#if PONY_LLVM >= 600
678:#if PONY_LLVM < 600
740:#if PONY_LLVM == 600
802:#if PONY_LLVM < 500
823:#if PONY_LLVM < 500

src/libponyc/codegen/genbox.c
34:#if PONY_LLVM >= 400
67:#if PONY_LLVM >= 400

src/libponyc/codegen/host.cc
50:#if PONY_LLVM < 600
79:#if PONY_LLVM < 500 and defined(PLATFORM_IS_X86)
102:#if PONY_LLVM < 500 and defined(PLATFORM_IS_X86)
112:#if PONY_LLVM < 600
234:#if PONY_LLVM < 500
247:#if PONY_LLVM < 500

src/libponyc/codegen/codegen.h
12:#define PONY_LLVM ((LLVM_VERSION_MAJOR * 100) + LLVM_VERSION_MINOR)

src/libponyc/codegen/genopt.cc
35:#if PONY_LLVM >= 600
95:#if PONY_LLVM == 600
199:#if PONY_LLVM < 500
210:#if PONY_LLVM < 400
322:#if PONY_LLVM >= 400 && !defined(_MSC_VER)
879:#if PONY_LLVM < 500
891:#if PONY_LLVM < 500
902:#if PONY_LLVM < 500
1292:#if PONY_LLVM < 500
1309:#if PONY_LLVM < 500
1376:#if PONY_LLVM < 500
1525:#if PONY_LLVM >= 400
1538:#if PONY_LLVM >= 400
1555:#if PONY_LLVM >= 400

src/libponyc/codegen/gentype.c
23:#if PONY_LLVM >= 600
163:#if PONY_LLVM < 400
166:  return LLVMMDNodeInContext(ctx, params, PONY_LLVM < 400 ? 3 : 2);
398:#if PONY_LLVM < 600

src/libponyc/codegen/gendebug.h
6:#if PONY_LLVM >= 600
52:#if PONY_LLVM < 600

src/libponyc/codegen/genreference.c
119:#if PONY_LLVM >= 400

src/libponyc/codegen/genprim.c
224:#if PONY_LLVM >= 400
264:#if PONY_LLVM >= 400
375:#if PONY_LLVM >= 400

src/libponyc/codegen/gencall.c
345:#if PONY_LLVM >= 400

test/libponyc/codegen.cc
466:#if !defined(PLATFORM_IS_WINDOWS) && (PONY_LLVM >= 500)
```

Reply:
See comment below. We don't want to support multiple llvms. We haven't finalized what we want to do with cleaning info out of the codebase. I asusme, yes we would drop all the above and perhaps tag the last version before we removed so if we have for reference.

Reply:
I don't think we should remove though as part of this commit. I think it makes sense to do in a follow on commit.